### PR TITLE
Add fingerprint() and key_id() method

### PR DESCRIPTION
**Pull request by chocol4te**
I have added test outlines, but am unsure of how you would like the tests to be structured. These functions should apply to both a whole parsed key, but also apply to subkeys.

I tested only with DSA and RSA on V3 and V4, and *should* be correct for other algorithms.

### CODE REVIEW

File: src/packet/types/key.rs [hunk 1]
```diff
@@ -257,11 +261,137 @@ macro_rules! key {
...
+                        }
+
+                        h.update(&packet).unwrap();
+
+                        h.finish().unwrap().deref().to_vec()
+                    }
+                }
+            }
+
+            pub fn key_id(&self) -> Vec<u8> {
```
Comment (dignifiedquire) [inline src/packet/types/key.rs]: given that this is not always supported I think returning `Option<Vec>` would be better

File: src/composed/key.rs [hunk 2]
```diff
@@ -555,4 +565,39 @@ mod tests {
 
         assert_eq!(ua.signatures, vec![sig3]);
     }
+
+    #[test]
+    fn test_fingerprint_rsa() {
+        // V4
```
Comment (dignifiedquire) [inline src/composed/key.rs]: The testcases in `test/opengpg-interop/testcases/keys` have json files that contain the right fingerprints.  I plan to parse and test them automatically, but for now you can just hardcode these from there. A single key for each is good enough for now

File: src/packet/types/key.rs (hunk 1 above)
Comment (chocol4te) [reply src/packet/types/key.rs]: What would result in returning None? I can't find much on the specification :/

File: src/composed/key.rs (hunk 2 above)
Comment (chocol4te) [reply src/composed/key.rs]: Sure thing, writing them now :)

File: src/packet/types/key.rs (hunk 1 above)
Comment (dignifiedquire) [reply src/packet/types/key.rs]: if it’s a v3 key with non rsa algorithm, though, instead of an empty vec

File: src/composed/key.rs [hunk 3]
```diff
@@ -301,7 +314,10 @@ mod tests {
         let input = ::std::str::from_utf8(raw).unwrap();
         let key = PublicKey::from_string(input).expect("failed to parse key");
 
-        // assert_eq!(key.primary_key.fingerprint(), "56c65c513a0d1b9cff532d784c073ae0c8445c0c");
+        assert_eq!(
+            hex::encode(key.primary_key.fingerprint()),
+            "56c65c513a0d1b9cff532d784c073ae0c8445c0c"
```
Comment (dignifiedquire) [inline src/composed/key.rs]: there was already hex_literal used in another test, can you replace that with this as well, no need to have to crated for hex decoding and this one looks nicer

Comment (chocol4te) [reply src/composed/key.rs]: Will do :)

**Comment by chocol4te** [comment]
I think I just have one last bug to fix for ECDH keys and this should be good to go.

**Pull request merged**